differ/mutationDiffer.go

```diff
@@ -756,79 +764,89 @@ func (dw *DifferWorker) diff() {
 					continue
 				}
-				if !isKeyNotFoundError(sourceResult.Error()) && isKeyNotFoundError(targetResult.Error()) {
+				if !isKeyNotFoundError(srcerr) && isKeyNotFoundError(tgterr) {
 					if _, exists := missingFromTarget[tgtColId]; !exists {
 						missingFromTarget[tgtColId] = make(map[string]*GocbResult)
 					}
-					missingFromTarget[tgtColId][key] = gocbResultConstructor(sourceResult.GoCbResult())
+					missingFromTarget[tgtColId][key] = gocbResultConstructor(sourceResult)
 					continue
 				}
-				if !areResultsTheSame(sourceResult.GoCbResult(), targetResult.GoCbResult()) {
-					if isDeletedPerMetadata != nil && isDeletedPerMetadata(sourceResult.GoCbResult()) {
-						if _, exists := deletedFromSource[srcColId]; !exists {
-							deletedFromSource[srcColId] = make(map[string][]*GocbResult)
+				if bodyOnly {
+					if !areGetResultsBodyTheSame(sourceResult, targetResult) {
+						if _, exists := srcDiff[srcColId]; !exists {
+							srcDiff[srcColId] = make(map[string][]*GocbResult)
 						}
-						deletedFromSource[srcColId][key] = append(deletedFromSource[srcColId][key], []*GocbResult{gocbResultConstructor(sourceResult.GoCbResult()), gocbResultConstructor(targetResult.GoCbResult())}...)
-						continue
-					}
-					if isDeletedPerMetadata != nil && isDeletedPerMetadata(targetResult.GoCbResult()) {
-						if _, exists := deletedFromTarget[tgtColId]; !exists {
-							deletedFromTarget[tgtColId] = make(map[string][]*GocbResult)
+						srcDiff[srcColId][key] = append(srcDiff[srcColId][key], []*GocbResult{gocbResultConstructor(sourceResult), gocbResultConstructor(targetResult)}...)
+						if _, exists := tgtDiff[tgtColId]; !exists {
+							tgtDiff[tgtColId] = make(map[string][]*GocbResult)
 						}
-						deletedFromTarget[tgtColId][key] = append(deletedFromTarget[tgtColId][key], []*GocbResult{gocbResultConstructor(sourceResult.GoCbResult()), gocbResultConstructor(targetResult.GoCbResult())}...)
-						continue
+						tgtDiff[tgtColId][key] = append(tgtDiff[tgtColId][key], []*GocbResult{gocbResultConstructor(targetResult), gocbResultConstructor(sourceResult)}...)
 					}
-					if _, exists := srcDiff[srcColId]; !exists {
-						srcDiff[srcColId] = make(map[string][]*GocbResult)
+				} else {
+					srcUUID, err := hlv.UUIDtoDocumentSource(dw.differ.sourceBucketUUID)
+					if err != nil {
+						dw.logger.Errorf("Error in converting the bucket UUID to required HLV format. err: %v", err)
+						panic("bucketUUID convertion error")
 					}
-					srcDiff[srcColId][key] = append(srcDiff[srcColId][key], []*GocbResult{gocbResultConstructor(sourceResult.GoCbResult()), gocbResultConstructor(targetResult.GoCbResult())}...)
-					if _, exists := tgtDiff[tgtColId]; !exists {
-						tgtDiff[tgtColId] = make(map[string][]*GocbResult)
+					tgtUUID, err1 := hlv.UUIDtoDocumentSource(dw.differ.targetBucketUUID)
+					if err1 != nil {
+						dw.logger.Errorf("Error in converting the bucket UUID to required HLV format. err: %v", err1)
```

Comment:
Same log document key

Reply:
Similarly have moved this also outside the for loop.
In my opinion we don't need to log the document key here and instead logging the input bucket UUID

Reply:
I'm not understanding the above comment. What is outside the for loop? And why you think we don't need to log document key?

Reply:
Initially in the `MutationDiffer.Diff()`method I was  converting the src and tgt bucket UUIDs to base 64 for every key in the fetch list.
So now i convert the source and target bucket UUIDs to base64 even before starting to iterate over the fetchList keys. Hence I do not the see the need to log the document key.